### PR TITLE
Sync search params with URL

### PR DESCRIPTION
**Pull request by fhlavac**
https://issues.redhat.com/browse/HAC-1433

Added search params sync with the URL for the table component so that search params change in the URL is reflected in search bar and vice versa.

syncURL

### CODE REVIEW

**Review by fhlavac — COMMENTED** [review]
@vojtechszocs just not sure about this - it's probably eslint false positive
https://github.com/openshift/dynamic-plugin-sdk/pull/108/files#diff-d7b35d11b677828ab3332b86d539dfcc0c092a5afd51e65778371f871676929aR64
https://github.com/openshift/dynamic-plugin-sdk/pull/108/files#diff-d7b35d11b677828ab3332b86d539dfcc0c092a5afd51e65778371f871676929aR84

File: packages/lib-utils/src/utils/url-sync.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    }
+  });
+
+  history.replace({
+    pathname: history.location.pathname,
+    search: searchParams.toString(),
+  });
+};
+
+export const syncFiltersWithUrl = (
```
Comment (vojtechszocs) [inline packages/lib-utils/src/utils/url-sync.ts:22]:
```suggestion
export const syncFiltersWithURL = (
```

File: packages/lib-utils/src/utils/url-sync.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  keys: string[],
+  defaults: Record<string, string> = {},
+) => {
+  const searchParams = new URLSearchParams(history.location.search);
+
+  let filters: Record<string, string> = keys.reduce((acc, key) => {
+    const values = searchParams.getAll(key);
+    return {
+      ...acc,
+      [key]: values.length > 1 ? values : values[0],
```
Comment (florkbr) [inline packages/lib-utils/src/utils/url-sync.ts:33]: @fhlavac I wonder if we should update the logic below to handle the array in either scenario you're covering here - that way we are consistent and don't need as may `.length` checks?

File: packages/lib-utils/src/utils/url-sync.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    };
+  }, {});
+
+  Object.keys(defaults).forEach((key: string) => {
+    const value = defaults[key];
+    filters = {
```
Comment (florkbr) [inline packages/lib-utils/src/utils/url-sync.ts:39]: @fhlavac is the only thing we're doing here is reassigning the value of `filters[key]` here to a default value if it's not set? We might want to favor chaining for readability - and we could also merge the objects together (defaults and parsed filters) using [lodash merge](https://lodash.com/docs/#merge). Something like: `_.merge(defaults, filters)`

**Comment by fhlavac** [comment]
@vojtechszocs updated the PR according to your suggestions, works much better. Thank you!

**Pull request merged**